File: Java/DesignPatterns/src/test/java/designPatterns/behavioral/strategy/MainStrategyTest.java
package designPatterns.behavioral.strategy;

import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

public class MainStrategyTest {
  private Context context;

  @Test
  void equalsTest() {
    context = new Context(new Operation.Add());
    assertThat(context.executeStrategy(10, 5)).isEqualTo(10 + 5);

    context = new Context(new Operation.Subtract());
    assertThat(context.executeStrategy(10, 5)).isEqualTo(10 - 5);

    context = new Context(new Operation.Multiply());
    assertThat(context.executeStrategy(10, 5)).isEqualTo(10 * 5);
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/behavioral/strategy/MainStrategyTest.kt
package behavioral.strategy

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class MainStrategyTest {
  private lateinit var context: Context

  @Test
  internal fun equalsTest() {
    context = Context(Operation.Add())
    assertThat(context.executeStrategy(10, 5)).isEqualTo(10 + 5)

    context = Context(Operation.Subtract())
    assertThat(context.executeStrategy(10, 5)).isEqualTo(10 - 5)

    context = Context(Operation.Multiply())
    assertThat(context.executeStrategy(10, 5)).isEqualTo(10 * 5)
  }
}